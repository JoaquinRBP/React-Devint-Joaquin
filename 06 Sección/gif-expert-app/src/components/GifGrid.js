import React from 'react';
import { GifGridItem } from './GifGridItem';
import {useFetchGifs} from '../hooks/useFetchGifs';
//import React, {useState, useEffect} from 'react';
//import {getGifs} from './../helpers/getGifs';
export const GifGrid = ({category})=>{
    //const [Count,setCount] = useState(0);
    // const [images, setimages] = useState([])
    // useEffect(()=>{
    //     getGifs(category)
    //     .then(setimages)
    // }, [category]) //segundo arreglo es una lista de dependencias
    const {data: imagenes,loading} = useFetchGifs(category)
    
    return (
        <>
        <h3>{category}</h3>
        {loading && <p className="animate__animated animate__flash">Loading</p>}
        <div className="card-grid">
            <ol>{
                imagenes.map(img =>(
                    <GifGridItem 
                    key={img.id}
                    //img={img}
                    {...img}
                    />
                ))}
            </ol>
        </div>
        </>
    )
}