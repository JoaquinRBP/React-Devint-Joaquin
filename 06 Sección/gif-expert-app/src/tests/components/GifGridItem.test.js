import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';
describe('Pruebas sobre <GifGridItem />',()=>{
    const title = 'Pruebas sobre';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem url={url} title={title}/>);
    test('Pruebas sobre <GifGridItem />',()=>{
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de tener un parrafo con el title',()=>{
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    test('debe de tener la imagen igual al url',()=>{
        const img = wrapper.find('img');
        //expect(img.props().src).toBe(url);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
    test('debe tener animate__fadeIn',()=>{
        const div = wrapper.find('div');
        const className=div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
})