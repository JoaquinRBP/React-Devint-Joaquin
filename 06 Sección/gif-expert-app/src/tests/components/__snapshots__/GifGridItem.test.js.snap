// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGridItem /> Pruebas sobre <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Pruebas sobre"
    src="https://localhost/algo.jpg"
  />
  <p>
    Pruebas sobre
  </p>
</div>
`;
